src/ray/object_manager/object_manager.cc

```diff
@@ -478,28 +527,24 @@ void ObjectManager::Push(const ObjectID &object_id, const NodeID &client_id) {
                    << ", total data size: " << data_size;
 
     UniqueID push_id = UniqueID::FromRandom();
-    for (uint64_t chunk_index = 0; chunk_index < num_chunks; ++chunk_index) {
-      rpc_service_.post([this, push_id, object_id, owner_address, client_id, data_size,
-                         metadata_size, chunk_index, rpc_client]() {
-        auto st = SendObjectChunk(push_id, object_id, owner_address, client_id, data_size,
-                                  metadata_size, chunk_index, rpc_client);
-        if (!st.ok()) {
-          RAY_LOG(WARNING) << "Send object " << object_id << " chunk failed due to "
-                           << st.message() << ", chunk index " << chunk_index;
-        }
-      });
-    }
+    push_manager_->StartPush(push_id, num_chunks, [=](int64_t chunk_id) {
+      SendObjectChunk(push_id, object_id, owner_address, client_id, data_size,
+                      metadata_size, chunk_id, rpc_client,
+                      [=](const Status &status) { push_manager_->OnChunkComplete(); });
```

Comment:
Don't we need any logic when `status != Status::OK()`?

Reply:
I have a TODO comment to handle, not needed for now.